### PR TITLE
test: add new test code (app_test) & base client

### PR DESCRIPTION
**Pull request by zheng1**
Generate client code with `goswagger`.

### CODE REVIEW

File: test/app_test.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package test
+
+import (
+	"openpitrix.io/openpitrix/test/client/app_service"
+	"openpitrix.io/openpitrix/test/models"
+	"log"
+	"testing"
+	"github.com/go-openapi/runtime"
```
Comment (chilianyi) [inline test/app_test.go]: The order should be standard, other, our package, so here should be sorted. 
And then go fmt, indent is 1 tab=8 space

File: test/client/app_runtime_service/app_runtime_service_client.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
+// Code generated by go-swagger; DO NOT EDIT.
+
+package app_runtime_service
```
Comment (chilianyi) [inline test/client/app_runtime_service/app_runtime_service_client.go]: I saw the code nearly the same as app_runtime.pb.go. 
Can we just use app_runtime.pb.go in app_test.go to make e2e test?

File: test/app_test.go (hunk 1 above)
Comment (zheng1) [reply test/app_test.go]: i saw the editorconfig is:
```plain
[*.{go,proto}]
charset = utf-8
indent_style = tab
tab_width = 4
```

File: test/client/app_runtime_service/app_runtime_service_client.go (hunk 2 above)
Comment (zheng1) [reply test/client/app_runtime_service/app_runtime_service_client.go]: I think this file is test all of `api server`, but `app_runtime.pb.go` is only `app server`.

File: test/app_test.go (hunk 1 above)
Comment (chilianyi) [reply test/app_test.go]: Sorry, I was misremembered.

File: test/client/app_runtime_service/app_runtime_service_client.go (hunk 2 above)
Comment (chilianyi) [reply test/client/app_runtime_service/app_runtime_service_client.go]: Ok, I know, `app_runtime.pb.go` can only test `grpc`, but this file can test `http`.

File: test/Makefile [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+.PHONY: generate
+generate:
+	cat ../pkg/cmd/api/spec/*.swagger.json | jq --slurp 'reduce .[] as $$item ({}; . * $$item)' > api.swagger.json
+	swagger generate client -f api.swagger.json -A openpitrix
+	rm api.swagger.json
+
+.PHONY: test
+test:
+	go test
```
Comment (chai2010) [inline test/Makefile]: if the local openpitrix server missing, the go test will failed.
maybe we should skip the test as default.
for developer, we hope `go test ./...` is as fast as `go build ./...`.
if we need do more test, we could enable the more test by command line flags.

File: test/app_test.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+	// delete old app
+	testAppId := "app-xxxxxxxy"
+	testAppName := "test1"
+	deleteAppParams := app_service.NewDeleteAppParams()
+	deleteAppParams.SetID(testAppId)
+	deleteAppRet, err := client.AppService.DeleteApp(deleteAppParams)
+	if err != nil {
+		if serr, ok := err.(*runtime.APIError); ok {
+			if resp, ok := serr.Response.(runtime.ClientResponse); ok {
+				log.Fatal(resp.Body())
```
Comment (chai2010) [inline test/app_test.go]: if it is in `go test`, it should be `t.Fatal`

File: test/app_test.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	appModel := models.OpenpitrixApp{ID: testAppId, Name: testAppName}
+	createAppParams := app_service.NewCreateAppParams()
+	createAppParams.WithBody(&appModel)
+	createAppRet, err := client.AppService.CreateApp(createAppParams)
+	if err != nil {
+		if serr, ok := err.(*runtime.APIError); ok {
+			if resp, ok := serr.Response.(runtime.ClientResponse); ok {
+				log.Fatal(resp.Body())
+			}
+			log.Fatal(serr.Response)
```
Comment (chai2010) [inline test/app_test.go]: ditto

File: test/app_test.go [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+	getAppRet, err := client.AppService.GetApp(getAppParams)
+	if err != nil {
+		log.Fatal(err)
+	}
+	app := getAppRet.Payload
+	if app == nil {
+		t.Errorf("failed to get app [%s]", testAppId)
+	}
+	log.Print(getAppRet)
+	t.Log("test app finish")
```
Comment (chai2010) [inline test/app_test.go]: no message is ok for go test

**Pull request merged**